artiq/coredevice/comm_kernel.py

```diff
@@ -565,6 +565,8 @@ def service(obj, attr, value): return setattr(obj, attr, value)
             self._flush()
         except RPCReturnValueError as exn:
             raise
+        except struct.error as exn:
+            raise RPCReturnValueError(f"cannot serialize data {result} into bytes, exception: {exn}")
```

Comment:
Use ``raise from``, not ``exception: {exn}``.
Why is ``RPCReturnValueError`` the correct exception type to use here?